x/pricefeed/keeper/keeper.go

```diff
@@ -111,40 +119,38 @@ func (k Keeper) SetCurrentPrices(ctx sdk.Context, marketID string) error {
 			notExpiredPrices = append(notExpiredPrices, types.NewCurrentPrice(v.MarketID, v.Price))
 		}
 	}
+
 	if len(notExpiredPrices) == 0 {
-		store := ctx.KVStore(k.key)
-		store.Set(
-			types.CurrentPriceKey(marketID), k.cdc.MustMarshalBinaryBare(types.CurrentPrice{}),
-		)
+		// TODO: why do we set an empty value?
+		k.setCurrentPrice(ctx, marketID, types.CurrentPrice{})
```

Comment:
The current price stored will continue storing the most recent (expired) price if this in not set. This zero's out the current price stored value for that market and ensures that CDP methods that `GetCurrentPrice` will return error.